Dockerfile

```diff
@@ -0,0 +1,23 @@
+# Build a Docker image for a Python application using Poetry
+
+# Set base image (host OS)
+FROM python:3.12-slim
+
+# Set environment variables
+ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONUNBUFFERED 1
+
+# Set work directory
+WORKDIR /app
+
+# Copy application code into the container
+COPY . /app
+
+# Install dependencies
+RUN pip install --upgrade pip && pip install poetry && poetry config virtualenvs.in-project true && poetry install --no-dev --no-root
+
+# Expose port for the application
+EXPOSE 8000
+
```

Comment:
[non-blocking] I think you should also have a line here that runs the migrations - it would be a good idea to have an `entrypoint.sh` script that handles all that. This can be added in a following PR when you make to change to PostgreSQL.